usermods/AHT10_v2/usermod_aht10.h

```diff
@@ -5,24 +5,22 @@
 
 #define AHT10_SUCCESS 1
 
-#define STATE_SETTING_ENABLED 0b00000001
-#define STATE_SETTING_MQTT_PUBLISH 0b00000010
-#define STATE_SETTING_MQTT_PUBLISHALWAYS 0b00000100
-#define STATE_SETTING_MQTT_HOMEASSISTANT 0b00001000
-#define STATE_INITDONE 0b00010000
-
-#define STATE_ISSET(stateFlags, state) (((stateFlags) & (state)) == (state))
-#define STATE_SET(stateFlags, state, set) ((set) ? ((stateFlags) |= (state)) : ((stateFlags) &= ~(state)))
-
 class UsermodAHT10 : public Usermod
 {
 private:
   static const char _name[];
 
   unsigned long _lastLoopCheck = 0;
 
-  // To avoid storing N bools, this flags field will handle all of those
-  uint8_t _stateFlags = 0;
+  struct
+  {
+    bool settingEnabled : 1;    // Enable the usermod
+    bool mqttPublish : 1;       // Publish mqtt values
+    bool mqttPublishAlways : 1; // Publish always, regardless if there is a change
+    bool mqttHomeAssistant : 1; // Enable Home Assistant docs
+    bool initDone : 1;          // Initialization is done
+    unsigned : 3;
+  } _stateFlags;
```

Comment:
FYI, there is no need to have struct object to use bitfields.

Reply:
Oh. Neat.

Reply:
Using these packed bits makes for way easier code. :)